semeio/jobs/scripts/fm_pyscal.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python
+"""Forward model for connecting ERT with the Pyscal command line client"""
+import logging
+import sys
+import os
+import six
+
+from semeio.jobs.design_kw.design_kw import extract_key_value
+
+from pyscal import pyscalcli
+
+_logger = logging.getLogger("FM_PYSCAL")
+
+MAGIC_NONE = "__NONE__"
+
+
+def main():
+    """As a forward model, all arguments are mandatory, but can have
+    defaults inserted"""
+    run(
+        sys.argv[1],
+        sys.argv[2],
+        sys.argv[3],
+        sys.argv[4],
+        sys.argv[5],
+        sys.argv[6],
+        sys.argv[7],
+    )
+
+
+def run(
+    relperm_parameters_file,
+    output_filename,  # make dir if non-existing
+    sheet_name,  # string, use 0 or __NONE__ when irrelevant or default
+    int_param_wo_name,  # string or __NONE__
+    int_param_go_name,  # string or __NONE__
+    slgof,  # sgof or slgof, default sgof
+    family,  # int: 1 or 2, default 1
+    parameters_file_name="parameters.txt",
+):
+    """This function is a wrapper around the Pyscal command
+    line tool. The command line tool is designed around argparse
+    and this function wraps around that design.
+
+    In contrast with the command line tool where interpolation
+    parameters are explicit, they are implicit here, and gathered
+    from parameters.txt"""
+    if not isinstance(relperm_parameters_file, six.string_types):
```

Comment:
This, and the following, checks should go into an argparse validator.

Reply:
Done